provider_client.go

```diff
@@ -389,7 +395,7 @@ func (client *ProviderClient) doRequest(method, url string, options *RequestOpts
 	}
 
 	// Construct the http.Request.
-	req, err := http.NewRequest(method, url, body)
+	req, err := http.NewRequestWithContext(ctx, method, url, body)
```

Comment:
Look we have a problem a couple lines below. If the Provider client contained a context, then that one will simply override this one. One solution could be to merge the two contexts with something like this: https://github.com/gophercloud/gophercloud/pull/2274/files#diff-552f47512a00afe5fc6850cc9ddc830a6daeca162750e50aab4ed549685e0253R28